### PR TITLE
Alchemy

### PR DESCRIPTION
**Pull request by Iamgoofball**
its-magic-i-aint-gotta-explain-shit2
TODO:
- make shit work
- fix web of lies slowdown always being active(wtf flag system)
- add more bases
- add mushrooms to grind down for bases
- make these mushrooms findable on mining and maint
- add shitloads more potions
- add shitloads more curses
- add ways to get alch equipment
- add ways to get bases
- fix bugs

### CODE REVIEW

**Comment by PKPenguin321** [comment]
Chemistry DLC: Magic Edition
Looks pretty neat so far, especially since it doesn't completely tear out the chemistry system we already have and adds onto it instead. :+1:
But seriously, this looks cool from what I can make of it.

**Comment by sawu-tg** [comment]
@Iamgoofball <sawu> also iamgoofball add a potion called "Dragon's Roll' that increase your speed and makes you *flip constantly for the duration

**Comment by Lumipharon** [comment]
if the away mission 2.0 thing ever gets off the ground, that would be a neat place to locate stuff for this.

**Comment by Iamgoofball** [comment]
Yeah probs


# PS GUYS SUGGEST POTION IDEAS, I ACCEPT ANYTHING, REMEMBER ITS MAGIC AND WE DONT HAVE TO EXPLAIN SHIT, IT CAN BE POSITIVE, NEGATIVE OR NEITHRR

**Comment by fj45** [comment]
Can't forget the classic-
Alchemical fire.

**Comment by Iamgoofball** [comment]
Also probably need some actual full on "make shit into gold" since it's alchemy

**Comment by kyrahabattoir** [comment]
need RNGzus

**Comment by Dorsisdwarf** [comment]
This seems pretty dumb, but really who cares.
oh and make russula Powder come from grinding dried mushrooms, since russulas are a type of mushroom

**Comment by Lumipharon** [comment]
mana pots for wizards. Instantly recharges all their spells.
Damage resist/fire resist pots etc

**Comment by Alek2ander** [comment]
Gold items? #8380 attempt 2 go.

**Comment by Alek2ander** [comment]
>Berserker's rage not reducing stuns

**Comment by Firecage** [comment]
Resist [element] potions? Fire for...well, fires. Cold for...well...when its cold. Shock for electricity?

**Comment by xxalpha** [comment]
full metal goofchemist

**Comment by MrPerson** [comment]
I am absolutely against this 1000%. They clash violently with the theme of the rest of the game. The ingredients aren't even interesting to acquire. OOOH BLOOD however will I acquire blood? Orange juice, oh jesus, don't break my legs.

File: code/modules/reagents/Chemistry-Holder.dm [hunk 1]
```diff
@@ -410,6 +411,20 @@ var/const/INGEST = 2
...
+		else
+			M.status_flags &= ~INCREASEDAMAGE
+
+/datum/reagents/proc/check_slowdown(var/mob/M)
+	if(istype(M, /mob))
+		if(M.reagents.has_reagent("web_of_lies"))
+			return 1
+		else
+			M.status_flags &= ~SLOWDOWN
+
```
Comment (MrPerson) [inline code/modules/reagents/Chemistry-Holder.dm]: This is all shit

File: code/modules/reagents/Chemistry-Holder.dm [hunk 2]
```diff
@@ -398,7 +399,7 @@ var/const/INGEST = 2
 
 /datum/reagents/proc/check_gofast(var/mob/M)
 	if(istype(M, /mob))
-		if(M.reagents.has_reagent("unholywater")||M.reagents.has_reagent("nuka_cola"))
+		if(M.reagents.has_reagent("unholywater")||M.reagents.has_reagent("nuka_cola")||M.reagents.has_reagent("endurance_feat"))
```
Comment (MrPerson) [inline code/modules/reagents/Chemistry-Holder.dm]: Oh goodie you're adding more stuff into the worst part of your shitcode, yay!

File: code/modules/reagents/Chemistry-Reagents/alchemy/Bases.dm [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+	description = ""
+	reagent_state = SOLID
+	color = "#673910" // rgb: 103, 57, 16
+
+/datum/reagent/alchemy/psathyrella_powder
+	name = "Psathyrella Powder"
+	id = "psathyrella_powder"
+	description = ""
+	reagent_state = SOLID
+	color = "#673910" // rgb: 103, 57, 16
```
Comment (MrPerson) [inline code/modules/reagents/Chemistry-Reagents/alchemy/Bases.dm]: I'm no expert, but I don't think any of these are acquirable in-game.

Comment (Iamgoofball) [inline code/modules/reagents/Chemistry-Reagents/alchemy/Bases.dm]: see: todo list in the description

File: code/modules/reagents/Chemistry-Holder.dm (hunk 2 above)
Comment (Iamgoofball) [inline code/modules/reagents/Chemistry-Holder.dm]: i absolutely hate these procs and I'm gonna adjust this shit

File: code/modules/reagents/Chemistry-Holder.dm (hunk 1 above)
Comment (Iamgoofball) [inline code/modules/reagents/Chemistry-Holder.dm]: the proc checking system is shit, yes

**Comment by KorPhaeron** [comment]
@MrPerson 

We already have wizards, cults and their gods, golems, flying fish, etc

We don't really have a straight sci fi theme.

@Iamgoofball 

Please use the ectoplasm item so I'm less embarrassed about it existing with no purpose other than ghost burgers

**Comment by Miauw62** [comment]
its a science station though
and without looking i can tell that these are shit because it says
"iamgoofball" as the author.

**Comment by xxalpha** [comment]
Harsh.

I disagree with this also, unless it's purely for the wizards, and in that case you might as well just spawn the potions instead of brewing them.

**Comment by duncathan** [comment]
There was talk of giving this to librarian. Given his occult desk it would fit him pretty well. Also he desperately needs gameplay other than RP.

**Comment by Miauw62** [comment]
making the librarian a magical chemist is a bad idea.
1. there are already chemists
2. the librarian will just do chemistry all the time

honestly just remove librarian, make the library public-access and replace his office with a reporter office.

**Comment by xxalpha** [comment]
>Given his occult desk it would fit him pretty well.

That doesn't make any sense and the librarian should not be an assistant with super powers. The librarian should be removed / have its workplace severely reduced. It's arguably the most uninteresting job right now, because nobody reads books.

**Comment by Iamgoofball** [comment]
I'll just do a poll and go from there

**Comment by AlexanderUlanH** [comment]
If this doesn't get in, can we at least get the throwing bottles and the slowdown/speedup mechanic changes?

**Comment by Razharas** [comment]
@Cheridan 
Your opinion on this is needed

**Comment by Iamgoofball** [comment]
This probably isn't getting in

**Pull request closed without merging**